go/exorelay/main.go

```diff
@@ -57,3 +66,42 @@ func (exoRelay *ExoRelay) Send(eventName string, payload map[string]interface{})
 
 	return websocket.Message.Send(exoRelay.socket, serializedBytes)
 }
+
+func (exoRelay *ExoRelay) receiveMessage(message structs.Message) {
+	if exoRelay.messageHandlerMapping == nil {
+		return
+	}
+	if exoRelay.messageHandlerMapping[message.Name] == nil {
+		return
+	}
+	exoRelay.messageHandlerMapping[message.Name](message.Payload)
+}
+
+func (exoRelay *ExoRelay) listenForMessages() {
+	for {
+		message, err := exoRelay.readMessage()
+		if err == io.EOF {
+			break
+		} else if err != nil {
+			fmt.Println("Error reading message from websocket:", err)
+			break
+		} else {
+			exoRelay.receiveMessage(message)
+		}
+	}
+}
+
+func (exoRelay *ExoRelay) readMessage() (structs.Message, error) {
+	var bytes []byte
+	if err := websocket.Message.Receive(exoRelay.socket, &bytes); err != nil {
+		return structs.Message{}, err
+	}
+
+	var unmarshaled structs.Message
+	err := json.Unmarshal(bytes, &unmarshaled)
+	if err != nil {
+		return structs.Message{}, err
+	}
+
+	return unmarshaled, nil
+}
```

Comment:
As a potential way to fix this, thoughts on bringing the `exocom-mock` (and `structs`) into the `exorelay` folder. Unless they need to be their own project, it would be easier to develop if they all were under `exorelay`.